frontend_app_caldavzap/src/container/Caldavzap.jsx

```diff
@@ -0,0 +1,141 @@
+import React from 'react'
+import { translate } from 'react-i18next'
+import i18n from '../i18n.js'
+import {
+  addAllResourceI18n,
+  handleFetchResult
+} from 'tracim_frontend_lib'
+import { debug } from '../helper.js'
+import { getCalendarList } from '../action.async.js'
+
+require('../css/index.styl')
+
+class Caldavzap extends React.Component {
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      appName: 'caldavzap',
+      isVisible: true,
+      config: props.data ? props.data.config : debug.config,
+      loggedUser: props.data ? props.data.loggedUser : debug.loggedUser,
+      content: props.data ? props.data.content : debug.content,
+      userWorkspaceList: [],
+      userWorkspaceListLoaded: false
+    }
+
+    // i18n has been init, add resources from frontend
+    addAllResourceI18n(i18n, this.state.config.translation, this.state.loggedUser.lang)
+    i18n.changeLanguage(this.state.loggedUser.lang)
+
+    document.addEventListener('appCustomEvent', this.customEventReducer)
+  }
+
+  customEventReducer = ({ detail: { type, data } }) => { // action: { type: '', data: {} }
+    const { state } = this
+
+    switch (type) {
+      case 'caldavzap_showApp':
+        console.log('%c<Caldavzap> Custom event', 'color: #28a745', type, data)
+        if (data.config.appConfig.idWorkspace !== state.config.appConfig.idWorkspace) {
+          this.loadCalendarList(data.config.appConfig.idWorkspace)
+        }
+        break
+      default:
+        break
+    }
+  }
+
+  componentDidMount () {
+    console.log('%c<Caldavzap> did mount', `color: ${this.state.config.hexcolor}`)
+    document.getElementById('appFullscreenContainer').style.flex = '1'
+
+    this.loadCalendarList(this.state.config.appConfig.idWorkspace)
+  }
+
+  componentDidUpdate (prevProps, prevState) {
+    const { state } = this
+
+    console.log('%c<Caldavzap> did update', `color: ${state.config.hexcolor}`, prevState, state)
+
+    if (prevState.config.appConfig.idWorkspace !== state.config.appConfig.idWorkspace) {
+      this.caldavzapIframe.contentWindow.location.reload()
+    }
+  }
+
+  loadCalendarList = async idWorkspace => {
+    const { state } = this
+
+    const fetchResultUserWorkspace = await handleFetchResult(
+      await getCalendarList(state.config.apiUrl, idWorkspace)
+    )
+
+    switch (fetchResultUserWorkspace.apiResponse.status) {
+      case 200:
+        this.setState(prev => ({
+          config: {
+            ...prev.config,
+            appConfig: {
+              ...prev.config.appConfig,
+              idWorkspace: idWorkspace
+            }
+          },
+          userWorkspaceList: fetchResultUserWorkspace.body,
+          userWorkspaceListLoaded: true
+        }))
+        break
+      case 400:
+        switch (fetchResultUserWorkspace.body.code) {
```

Comment:
why do you need a switch here ?

Reply:
to handle many potential error code from api that might only happen in case of a status http 400

Reply:
nothing to do here, I resolve